pkg/util/zookeeper_util.go

```diff
@@ -28,10 +29,22 @@ const (
 
 // Delete all znodes related to a specific Bookkeeper cluster
 func DeleteAllZnodes(bk *v1alpha1.BookkeeperCluster, pravegaClusterName string) (err error) {
-	host := []string{bk.Spec.ZookeeperUri}
+	zkUri := strings.Split(bk.Spec.ZookeeperUri, ":")
+	zkSvcName := ""
+	zkSvcPort := ""
+	if len(zkUri) >= 1 {
+		zkSvcName = zkUri[0]
+		if len(zkUri) == 1 {
+			zkSvcPort = "2181"
+		} else {
+			zkSvcPort = zkUri[1]
+		}
+	}
+	hostname := zkSvcName + "." + bk.Namespace + ".svc.cluster.local:" + zkSvcPort
```

Comment:
This fix assumes that Bookkeeper and Zookeeper always run in the same ns. It may be good to make that explicit in the documentation.